pkg/mdlsub/output_ddb.go

```diff
@@ -16,33 +16,36 @@ func init() {
 	outputFactories[OutputTypeDdb] = outputDdbFactory
 }
 
-func repoInit(config cfg.Config, logger mon.Logger, settings *SubscriberSettings) func(model interface{}) ddb.Repository {
-	return func(model interface{}) ddb.Repository {
-		repo := ddb.NewRepository(config, logger, &ddb.Settings{
+func repoInit(config cfg.Config, logger mon.Logger, settings *SubscriberSettings) func(model interface{}) (ddb.Repository, error) {
+	return func(model interface{}) (ddb.Repository, error) {
+		repo, err := ddb.NewRepository(config, logger, &ddb.Settings{
 			ModelId: settings.TargetModel,
 			Main: ddb.MainSettings{
 				Model:              model,
 				ReadCapacityUnits:  5,
 				WriteCapacityUnits: 5,
 			},
 		})
+		if err != nil {
+			return nil, fmt.Errorf("can not create ddb repository: %w", err)
+		}
 
-		return ddb.NewMetricRepository(config, logger, repo)
+		return ddb.NewMetricRepository(config, logger, repo), nil
 	}
 }
 
-func outputDdbFactory(config cfg.Config, logger mon.Logger, settings *SubscriberSettings, transformers VersionedModelTransformers) map[int]Output {
+func outputDdbFactory(config cfg.Config, logger mon.Logger, settings *SubscriberSettings, transformers VersionedModelTransformers) (map[int]Output, error) {
 	outputs := make(map[int]Output)
 
 	for version := range transformers {
```

Comment:
Does it make sense to already add the error to the different outputs?

Reply:
What do you mean?

Reply:
You're adding the error to the outputFactory, but are not passing anything from the actual output.